BizHawk.Client.EmuHawk/tools/Lua/Libraries/EmuLuaLibrary.Communication.cs

```diff
@@ -37,70 +33,129 @@ public static string GetLuaFunctionsList()
 			{
 				list.AppendLine(function.ToString());
 			}
-
 			return list.ToString();
 		}
 
 		[LuaMethod("socketServerScreenShot", "sends a screenshot to the Socket server")]
 		public string SocketServerScreenShot()
 		{
-			return GlobalWin.socketServer.SendScreenshot();
+			return GlobalWin.socketServer?.SendScreenshot();
 		}
+
 		[LuaMethod("socketServerScreenShotResponse", "sends a screenshot to the Socket server and retrieves the response")]
 		public string SocketServerScreenShotResponse()
 		{
-			return GlobalWin.socketServer.SendScreenshot(1000).ToString();
+			return GlobalWin.socketServer?.SendScreenshot(1000).ToString();
 		}
 
 		[LuaMethod("socketServerSend", "sends a string to the Socket server")]
 		public string SocketServerSend(string SendString)
 		{
-			return "Sent : " + GlobalWin.socketServer.SendString(SendString).ToString() + " bytes";
+			return "Sent : " + GlobalWin.socketServer?.SendString(SendString).ToString() + " bytes";

```

Comment:
Why not interpolate here as you did in ApiHawk? Also, don't you need to use conditional access in ApiHawk as well?

Reply:
For the record, "dunno" and "probably, maybe later" are acceptable answers

Reply:
Since I didn't implement ApiHawk _dunno_ is appropriate answer.